packages/builder/src/stores/portal/groups.ts

```diff
@@ -0,0 +1,93 @@
+import { get } from "svelte/store"
+import { API } from "@/api"
+import { licensing } from "@/stores/portal"
+import { UserGroup } from "@budibase/types"
+import { BudiStore } from "../BudiStore"
+
+class GroupStore extends BudiStore<UserGroup[]> {
+  constructor() {
+    super([])
+  }
+
+  updateStore = (group: UserGroup) => {
+    this.update(state => {
+      const currentIdx = state.findIndex(gr => gr._id === group._id)
+      if (currentIdx >= 0) {
+        state.splice(currentIdx, 1, group)
+      } else {
+        state.push(group)
+      }
+      return state
+    })
+  }
+
+  async init() {
+    // Only init if there is a groups license, just to be sure but the feature will be blocked
+    // on the backend anyway
+    if (get(licensing).groupsEnabled) {
+      const groups = await API.getGroups()
+      this.set(groups)
+    }
+  }
+
+  private async refreshGroup(groupId: string) {
+    const group = await API.getGroup(groupId)
+    this.updateStore(group)
+  }
+
+  async save(group: UserGroup) {
+    const { ...dataToSave } = group
+    delete dataToSave.scimInfo
```

Comment:
It's a real shame there's not a more elegant way to do this. I've seen `const { scimInfo, ...rest } = group` used in the past but the linter doesn't like it.